src/main/java/core/basesyntax/strategy/impl/BalanceStrategy.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.dto.Transaction;
+import core.basesyntax.strategy.Strategy;
+
+public class BalanceStrategy implements Strategy {
+    @Override
+    public void makeOperation(Transaction transaction) {
+        Storage.fruits.put(transaction.getFruit(), transaction.getValue());
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        return this.getClass() == obj.getClass();
+    }
```

Comment:
it is better to override equals() the classical way
the null check is absent here, and obj.getClass() will produce NPE
same for all strategies